### PR TITLE
🌱 E2E: Bump Kubernetes to v1.29.5

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

Bumping the Kubernetes versions used in e2e tests.
Upgrades are now from v1.28.5 to v1.29.5

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by jichenjc** [comment]
/lgtm

k8s 1.30 is out, not sure whether we want to catch that?

**Comment by lentzi90** [comment]
Yes I'll build images for that next. Just wanted to do it in step :sweat_smile:

**Comment by lentzi90** [comment]
I was a bit too fast triggering the tests... The image was still uploading :joy:

File: hack/ci/cloud-init/controller.yaml.tpl [hunk 1]
```diff
@@ -68,9 +68,9 @@
     # https://docs.openstack.org/glance/latest/admin/quotas.html
     IMAGE_URLS="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/amphora/2022-12-05/amphora-x64-haproxy.qcow2,"
     IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/cirros/2022-12-05/cirros-0.6.1-x86_64-disk.img,"
-    IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2023-09-29/ubuntu-2204-kube-v1.27.2.img,"
-    IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2024-01-10/ubuntu-2204-kube-v1.28.5.img,"
-    IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/flatcar/flatcar-stable-3815.2.0-kube-v1.28.5.img,"
+    IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2023-09-29/ubuntu-2204-kube-v1.28.5.img,"
+    IMAGE_URLS+="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2024-01-10/ubuntu-2204-kube-v1.29.5.img,"
```
Comment (EmilienM) [inline hack/ci/cloud-init/controller.yaml.tpl:72]: the URL doesn't work for me

Comment (lentzi90) [reply hack/ci/cloud-init/controller.yaml.tpl:72]: Stupid me tried to be quick and just set the new image name, forgetting that it is in another folder :facepalm:

**Comment by lentzi90** [comment]
@tormath1 is something needed for the sysext test? It has failed now twice so I suspect I have missed something :thinking:

**Pull request merged**